### PR TITLE
demonstrate ExtendedTracer

### PR DESCRIPTION
**Pull request by zeitlinger**
which was newly introduced in 1.33

### CODE REVIEW

File: manual-tracing/src/main/java/io/opentelemetry/example/tracing/ExampleConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  static OpenTelemetry initOpenTelemetry() {
+    SdkTracerProvider sdkTracerProvider =
+        SdkTracerProvider.builder()
+            .addSpanProcessor(SimpleSpanProcessor.create(new LoggingSpanExporter()))
+            .build();
+
+    OpenTelemetrySdk sdk =
+        OpenTelemetrySdk.builder()
+            .setTracerProvider(sdkTracerProvider)
+            .setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()))
```
Comment (breedx-splk) [inline manual-tracing/src/main/java/io/opentelemetry/example/tracing/ExampleConfiguration.java:37]: This is configured by default, isn't it?

File: manual-tracing/src/main/java/io/opentelemetry/example/tracing/ManualTracingExample.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    tracer = ExtendedTracer.create(openTelemetry.getTracer(INSTRUMENTATION_NAME));
+  }
+
+  public void myWonderfulUseCase() {
+    tracer.spanBuilder("calculate LLVM").startAndCall(this::calculateLlvm);
+  }
+
+  private String calculateLlvm() {
+    String type = "smarter";
+    Span.current().setAttribute("type", type);
```
Comment (breedx-splk) [inline manual-tracing/src/main/java/io/opentelemetry/example/tracing/ManualTracingExample.java:28]: I would inline `type` here -- no reason for it to be a local var.

**Review by breedx-splk — APPROVED** [review]
This is looking good to me. I think the http examples demonstrate the value here nicely, but I could _possibly_ see it being contentious to use the existing http example with the new incubating tracer.

In any case, nice work.

File: manual-tracing/src/main/java/io/opentelemetry/example/tracing/ExampleConfiguration.java (hunk 1 above)
Comment (zeitlinger) [reply manual-tracing/src/main/java/io/opentelemetry/example/tracing/ExampleConfiguration.java:37]: in auto-configure it is, but not in the builder, which is used here.

**Comment by zeitlinger** [comment]
> This is looking good to me. I think the http examples demonstrate the value here nicely, but I could _possibly_ see it being contentious to use the existing http example with the new incubating tracer.

I see the point. 
I went with this approach to avoid duplicating the example.

**Comment by zeitlinger** [comment]
@breedx-splk can you check again?

**Review by breedx-splk — APPROVED** [review]
Yup, still looking good. Thanks.

**Comment by zeitlinger** [comment]
@jack-berg can you merge?

File: manual-tracing/src/main/java/io/opentelemetry/example/tracing/ManualTracingExample.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    OpenTelemetry oTel = ExampleConfiguration.initOpenTelemetry();
+
+    // Start the example
+    ManualTracingExample example = new ManualTracingExample(oTel);
+    // Generate a few sample spans
+    for (int i = 0; i < 5; i++) {
+      example.myWonderfulUseCase();
+    }
+
+    example.setBaggage();
```
Comment (jack-berg) [inline manual-tracing/src/main/java/io/opentelemetry/example/tracing/ManualTracingExample.java:57]: Maybe accept a `Runnable` as an argument and rename to `setBaggageAndRun` to make it clear what the method does?

File: http/src/main/java/io/opentelemetry/example/http/HttpServer.java [hunk 4]
```diff
@@ -13,43 +13,24 @@
...
+import java.util.stream.Collectors;
 
 public final class HttpServer {
   // It's important to initialize your OpenTelemetry SDK as early in your application's lifecycle as
   // possible.
   private static final OpenTelemetry openTelemetry = ExampleConfiguration.initOpenTelemetry();
-  private static final Tracer tracer =
-      openTelemetry.getTracer("io.opentelemetry.example.http.HttpServer");
+  private static final ExtendedTracer tracer =
+      ExtendedTracer.create(openTelemetry.getTracer("io.opentelemetry.example.http.HttpServer"));
```
Comment (jack-berg) [inline http/src/main/java/io/opentelemetry/example/http/HttpServer.java:29]: Its arguably a bit strange to reference experimental APIs as part of canonical examples, but I think its fine if we're serious about incubating these to eventually incorporate into Tracer.

**Review by jack-berg — APPROVED** [review]

**Comment by zeitlinger** [comment]
@trask can you merge?

**Pull request merged**